services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableIndexManagerImpl.java

```diff
@@ -408,6 +413,34 @@ public long populateViewFromEntityReplication(final Long viewId, final ViewScope
 		// calculate the new CRC32;
 		return tableIndexDao.calculateCRC32ofTableView(viewId);
 	}
+
+	void validateMaxListLengthInAnnotationReplication(ObjectType objectType, long viewId, long viewTypeMask,

```

Comment:
We could move this at the dao since it's part of the validation logic of the copyEntityReplicationToView method (and should always apply for that method).